### PR TITLE
mysql mysql-client 8.0.30

### PR DESCRIPTION
**Pull request by yahonda**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by yahonda** [comment]
Cherry-pick the commit 62c8bb3a3e53cc1cd47602108df9e35c6c72069c from #106625 to update both mysql and mysql-client to 8.0.30 because mysql and mysql-client packages need to be the same version.

**Comment by yahonda** [comment]
Here is the link to MySQL 8.0.30 release note https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-30.html

File: Formula/mysql.rb [hunk 1]
```diff
@@ -86,6 +85,7 @@ def install
       -DWITH_UNIT_TESTS=OFF
       -DENABLED_LOCAL_INFILE=1
       -DWITH_INNODB_MEMCACHED=ON
+      -DWITH_ZLIB=bundled
```
Comment (carlocab) [inline Formula/mysql.rb:88]: Let's not do that. Instead, remove
```ruby
uses_from_macos "zlib"
```
and add
```ruby
depends_on "zlib" # Zlib 1.2.12+
```

**Review by carlocab — DISMISSED** [review]

File: Formula/mysql.rb (hunk 1 above)
Comment (yahonda) [reply Formula/mysql.rb]: Thanks for the suggestion.

**Review by carlocab — APPROVED** [review]
Thanks!

**Pull request closed without merging**